Controllers/UserController.cs

```diff
@@ -78,16 +67,15 @@ public async Task<IActionResult> Signup(RegisterDTO model)
                 return ApiBadRequest("Invalid Headers!");
             }
 
-            if (!(await _roleManager.RoleExistsAsync(model.RoleId.ToString())))
+            var user = new Employee
             {
-                await _roleManager.CreateAsync(new IdentityRole(model.RoleId.ToString()));
-            }
-
-
-            var user = new User
-            {
-                UserName = model.Username ?? model.Email,
+                Username = model.Username,
                 Email = model.Email,
+                Department = model.RoleId,
+                FirstName = "Test",
+                LastName = "Testing",
+                PersonalCode = "39000000000",
+                Status = EmployeeStatusId.Employed
             };
```

Comment:
Maybe we should move this user to seed?

Reply:
It's not part of seed it's part of user creation. We first need to update register dto to have all the fields. It's just hard coded values for required fields.